### DISCUSSION TITLE
Puppet 8 container needs to be restarted to register configuraiton changes in Gitlab

### DISCUSSION THREAD

**Discussion by timur-enet** [puppetlabs/community#59] [Troubleshooting] [answered]
Hello, we have recently upgraded Puppet from version 7 to 8 and we have encountered the following issue:

We use a containerized version of Puppet in conjunction with Gitlab where we store configuration data. Before upgrading to v8 the Puppet server used to acknowledge changes in Gitlab automatically (after performing a git pull via cron job)

Since we upgraded to v8 we have to restart the Puppet container in order for the managed systems to recieve the upgraded configuration.

We can not find any issues or irregularities in the log files and we do not know how to proceed. We are using docker image voxpupuli/container-puppetserver:main-8

Has anyone else encountered this issue before or has any hints on how to troubleshoot it?
Kind Regards
Timur

**Comment by bastelfreak** [comment, marked as answer]
Do probably don't clear the environment cache after deployments? https://github.com/voxpupuli/container-puppetserver?tab=readme-ov-file#note-about-environment-caching
